Public/Src/App/UnitTests/Bxl/ServerDeploymentTests.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using BuildXL;
+using BuildXL.Engine;
+using BuildXL.Utilities;
+using Test.BuildXL.TestUtilities.Xunit;
+using Xunit;
+
+namespace Test.Bxl
+{
+    public class ServerDeploymentTests : TemporaryStorageTestBase
+    {
+        [Fact]
+        public void TestMissingManifestDirectory()
+        {
+            string deploymentDir;
+            XAssert.IsTrue(ServerDeployment.IsServerDeploymentOutOfSync(TemporaryDirectory, null, out deploymentDir));
+        }
+
+        [Fact]
+        public void TestMissingManifestFile()
+        {
+            string manifestPath = Path.Combine(TemporaryDirectory, AppDeployment.DeploymentManifestFileName);
+
+            File.WriteAllText(
+                    Path.Combine(Path.GetDirectoryName(manifestPath), AppDeployment.DeploymentManifestFileName),

```

Comment:
Isn't `Path.GetDiretoryName(manifestPath)` always going to be the same as `TemporaryDirectory`?  

Furthermore, isn't this whole `Path.Combine(Path.GetDirectoryName(manifestPath), AppDeployment.DeploymentManifestFileName)` the same as `manifestPath`?

For better clarity, I'd write
```csharp
var manifestRootDir = TemporaryDirectory;
var manifestPath = Path.Combine(manifestRootDir, AppDeployment.DeploymentManifestFileName);
File.WriteAllText(manifestPath, ...);
...
```